core/domain/user_jobs_one_off.py

```diff
@@ -175,6 +175,24 @@ def reduce(key, stringified_values):
                 subscription_services.subscribe_to_collection(key, item['id'])
 
 
+class DashboardStatsOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for populating weekly dashboard stats for all registered
+    users who have a non-None value of UserStatsModel.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        if isinstance(item, user_models.UserSettingsModel):
+            user_services.save_last_dashboard_stats(item.id)
```

Comment:
update_dashboard_stats_log? I think that conveys what you're trying to do more clearly...